### PR TITLE
Add event return capability to Splunk returner

### PR DESCRIPTION
**Pull request by mymindstorm**
### What does this PR do?

Adds `event_returner` support to Splunk returner.

I've tested this patch against my company's Splunk environment.

### What issues does this PR fix or reference?
References: #50815

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by mymindstorm** [comment]
@dhiltonp is there anything I can do to have this PR reviewed?

File: salt/returners/splunk.py [hunk 1]
```diff
@@ -48,6 +48,14 @@ def returner(ret):
     return _send_splunk(ret)
 
 
+def event_return(events):
+    """
+    Return events to Splunk via the HTTP Event Collector
+    """
+    for event in events:
+        _send_splunk(event)
```
Comment (dhiltonp) [inline salt/returners/splunk.py:56]: Can you refactor `_send_splunk` to reduce the overhead of sending payloads? You can use the `batchEvent` if you want, but be sure to test it as I see a one or two bugs in it.

Here's what I suggest:

Take the 3 actions in `_send_splunk`: 

 - create an `http_event_collector`
 - preparing a payload
 - sending the payload through the event collector

Refactor the first 2 actions:

 - create a function that creates an event collector using opts (maybe `_event_collector(opts)`)
 - create a function that prepares a payload (maybe `_prepare_payload(event, opts)`)

Then:

1. Modify `returner` to be something like:

```python
def returner(ret):
    opts = _get_options()
    event_collector = _event_collector(opts)
    payload = _prepare_payload(ret, opts)
    event_collector.sendEvent(payload)
```

2. Modify `event_return` to be something like:

```python
def event_return(events):
    opts = _get_options()
    event_collector = _event_collector(opts)
    for event in events:
        payload = _prepare_payload(event, opts)
        event_collector.sendEvent(payload)
```

3. Delete `_send_splunk`.

**Review by dhiltonp — CHANGES_REQUESTED** [review]

File: salt/returners/splunk.py (hunk 1 above)
Comment (mymindstorm) [reply salt/returners/splunk.py]: Thank you for the detailed review. I've refactored as directed and removed the  `batchEvent` related code in `http_event_collector` as it is unused.

**Review by dhiltonp — APPROVED** [review]

File: salt/returners/splunk.py (hunk 1 above)
Comment (dhiltonp) [reply salt/returners/splunk.py]: Thanks, it looks great!

**Comment by dhiltonp** [comment]
re-run pr-fedora-35-x86_64-py3-pytest

**Comment by mymindstorm** [comment]
re-run pr-centos-7-x86_64-py3-pytest

**Comment by mymindstorm** [comment]
I updated the branch and the CI seems to be failing due to some problem communicating with AWS. Re-running...

**Comment by mymindstorm** [comment]
@dhiltonp the CI issues with this PR appear to be resolved.

**Pull request merged**